### PR TITLE
Display device geolocation

### PR DESCRIPTION
**Pull request by davidebriani**
Show a Geolocation tab in the Device page displaying the last known position of the device on a map, if available.

https://user-images.githubusercontent.com/60540759/142637928-3f7407a5-1a6d-4d8a-b2e0-f2e7d33789c0.mp4

### CODE REVIEW

File: frontend/public/index.html [hunk 1]
```diff
@@ -25,6 +25,12 @@
     <meta name="description" content="Edgehog" />
     <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
     <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+    <link
```
Comment (Pavinati) [inline frontend/public/index.html:28]: why importing it from a CDN compared to importing it from the npm package inside the Map component?
https://github.com/Leaflet/Leaflet/blob/master/dist/leaflet.css

Comment (davidebriani) [reply frontend/public/index.html:28]: Importing it via Webpack requires either a custom loaders configuration for Webpack or custom code to also import each image/asset file needed besides the CSS.
I think that linking it in the `index.html` is a simple and more maintainable approach for now.

File: frontend/src/components/Map.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+type Props = {
+  latitude: number;
+  longitude: number;
+  popup?: React.ReactNode;
+};
+
+const Map = ({ latitude, longitude, popup, ...props }: Props) => {
+  return (
+    <MapContainer
+      style={{ minHeight: 300 }}
```
Comment (Pavinati) [inline frontend/src/components/Map.tsx:30]: is it possible to use css classes instead of inline styling?

File: frontend/src/components/Icon.tsx [hunk 3]
```diff
@@ -40,6 +41,7 @@ const icons = {
   circle: faCircle,
   delete: faTrash,
   devices: faTabletAlt,
+  locationMarker: faMapMarkerAlt,
```
Comment (Pavinati) [inline frontend/src/components/Icon.tsx:44]: doesn't look like we're using this one yet

File: frontend/src/pages/Device.tsx [hunk 4]
```diff
@@ -193,6 +209,61 @@ const DeviceHardwareInfo = ({ deviceRef }: DeviceHardwareInfoProps) => {
...
+          />
+        </p>
+        <Map
+          latitude={location.latitude}
+          longitude={location.longitude}
+          popup={
+            <div>
+              <p>{location.address}</p>
+              <p>
+                {location.latitude}°N, {location.longitude}°E
```
Comment (szakhlypa) [inline frontend/src/pages/Device.tsx:257]: Latitudes and longitudes also can be negative.
Negative latitudes represent the southern hemisphere (so with latitude < 0 N -> S)
And negative longitudes represent the western hemisphere (so E -> W)

Comment (davidebriani) [reply frontend/src/pages/Device.tsx:257]: Nice catch. The convention tells to use either:
- a decimal representation with pure numbers
- a N/E/S/W representation with non-negative degrees
Let's stick with the same decimal format returned from the API for now.

File: frontend/src/components/Icon.tsx [hunk 5]
```diff
@@ -25,6 +25,7 @@ import {
   faCircle,
   faCompactDisc,
   faPlus,
+  faMapMarkerAlt,
```
Comment (Pavinati) [inline frontend/src/components/Icon.tsx:28]: you're still importing the image

**Review by Pavinati — APPROVED** [review]

**Review by bettio — APPROVED** [review]

**Pull request merged**